src/lib/eina/eina_thread.c

```diff
@@ -47,113 +47,7 @@
 #endif
 #endif
 
-static inline void *
-_eina_thread_join(Eina_Thread t)
-{
-   void *ret = NULL;
-   int err = pthread_join((pthread_t)t, &ret);
-
-   if (err == 0) return ret;
-   return NULL;
-}
-
-static inline Eina_Bool
-_eina_thread_create(Eina_Thread *t, int affinity, void *(*func)(void *data), void *data)
-{
-   int err;
-   pthread_attr_t attr;
-#ifndef _WIN32
-   sigset_t oldset, newset;
-#endif
-   
-   pthread_attr_init(&attr);
-   if (affinity >= 0)
-     {
-#ifdef EINA_HAVE_PTHREAD_AFFINITY
-        cpu_set_t cpu;
-
-        CPU_ZERO(&cpu);
-        CPU_SET(affinity, &cpu);
-        pthread_attr_setaffinity_np(&attr, sizeof(cpu), &cpu);
-#endif
-     }
-
-   /* setup initial locks */
-#ifndef _WIN32
-   sigemptyset(&newset);
-   sigaddset(&newset, SIGPIPE);
-   sigaddset(&newset, SIGALRM);
-   sigaddset(&newset, SIGCHLD);
-   sigaddset(&newset, SIGUSR1);
-   sigaddset(&newset, SIGUSR2);
-   sigaddset(&newset, SIGHUP);
-   sigaddset(&newset, SIGQUIT);
-   sigaddset(&newset, SIGINT);
-   sigaddset(&newset, SIGTERM);
-# ifdef SIGPWR
-   sigaddset(&newset, SIGPWR);
-# endif
-   pthread_sigmask(SIG_BLOCK, &newset, &oldset);
-#endif
-   err = pthread_create((pthread_t *)t, &attr, func, data);
-#ifndef _WIN32
-   pthread_sigmask(SIG_SETMASK, &oldset, NULL);
-#endif
-   pthread_attr_destroy(&attr);
-
-   if (err == 0) return EINA_TRUE;
-
-   return EINA_FALSE;
-}
-
-static inline Eina_Bool
-_eina_thread_equal(Eina_Thread t1, Eina_Thread t2)
-{
-   return pthread_equal((pthread_t)t1, (pthread_t)t2);
-}
-
-static inline Eina_Thread
-_eina_thread_self(void)
-{
-   return (Eina_Thread)pthread_self();
-}
-
-
-typedef struct _Eina_Thread_Call Eina_Thread_Call;
-struct _Eina_Thread_Call
-{
-   Eina_Thread_Cb func;
-   const void *data;
-
-   Eina_Thread_Priority prio;
-   int affinity;
-};
-
-static void *
-_eina_internal_call(void *context)
-{
-   Eina_Thread_Call *c = context;
-   void *r;
-   pthread_t self;
-
-   // Default this thread to not cancellable as per Eina documentation
-   eina_thread_cancellable_set(EINA_FALSE, NULL);
-
-   EINA_THREAD_CLEANUP_PUSH(free, c);
-
-   if (c->prio == EINA_THREAD_BACKGROUND ||
-       c->prio == EINA_THREAD_IDLE)
-     eina_sched_prio_drop();
-
-   self = pthread_self();
-   _eina_debug_thread_add(&self);
-   EINA_THREAD_CLEANUP_PUSH(_eina_debug_thread_del, &self);
-   r = c->func((void*) c->data, eina_thread_self());
-   EINA_THREAD_CLEANUP_POP(EINA_TRUE);
-   EINA_THREAD_CLEANUP_POP(EINA_TRUE);
-
-   return r;
-}
+static void *_eina_internal_call(void *context);
```

Comment:
Wouldn't better to put this at eina_thead.h?

Reply:
i'm not sure, because it is private method of eina_thread.c. I can remove this, and put the function before it is called